src/data_processing/wiki_to_netflix.py

```diff
@@ -0,0 +1,136 @@
+
+# Getting list of movie IDs, genre IDs, and director IDs from request
+def wiki_query(data_csv, user_agent):
+    wiki_movie_ids = []
+    wiki_genres = []
+    wiki_directors = []
+    
+    # csv_reader = list(csv.reader(open(data_csv)))
+    
+    for row in data_csv:
+        if row[1] != "NULL":
+            SPARQL = '''
+            SELECT * WHERE {
+                SERVICE wikibase:mwapi {
+                    bd:serviceParam wikibase:api "EntitySearch" ;
+                                    wikibase:endpoint "www.wikidata.org" ;
+                                    mwapi:search "%(Title)s" ;
+                                    mwapi:language "en" .
+                    ?item wikibase:apiOutputItem mwapi:item .
+                }
+
+                ?item wdt:P31/wdt:P279* wd:Q11424 .
+                
+                {
+                    # Get US release date
+                    ?item p:P577 ?releaseDateStatement .
+                    ?releaseDateStatement ps:P577 ?releaseDate .
+                    ?releaseDateStatement pq:P291 wd:Q30 .  
+                }
+                UNION
+                {
+                    # Get unspecified release date
+                    ?item p:P577 ?releaseDateStatement .
+                    ?releaseDateStatement ps:P577 ?releaseDate .
+                    FILTER NOT EXISTS { ?releaseDateStatement pq:P291 ?country }
+                }
+            
+                FILTER (YEAR(?releaseDate) = %(Year)d) .
+
+                ?item rdfs:label ?itemLabel .
+                FILTER (lang(?itemLabel) = "en") .
+
+                OPTIONAL {
+                    ?item wdt:P136 ?genre .
+                    ?genre rdfs:label ?genreLabel .
+                    FILTER (lang(?genreLabel) = "en") .
+                }
+
+                OPTIONAL {?item wdt:P57 ?director.
+                                ?director rdfs:label ?directorLabel.
+                                FILTER (lang(?directorLabel) = "en")}
+
+                SERVICE wikibase:label { bd:serviceParam wikibase:language "en" . }
+                }
+        
+            ''' % {'Title': row[2], 'Year': int(row[1])}
+
+            response = requests.post('https://query.wikidata.org/sparql',
+                      headers={'User-Agent': user_agent},
+                      data={
+                        'query': SPARQL,
+                        'format': 'json',
+                      }
+        )
+            response.raise_for_status() 
+            
+            data = response.json()
+            
+            wiki_movie_ids.append(wiki_feature_info(data, 'item'))
+            wiki_genres.append(wiki_feature_info(data, 'genreLabel'))
+            wiki_directors.append(wiki_feature_info(data, 'directorLabel'))
+        
+    return(wiki_movie_ids, wiki_genres, wiki_directors)
+
+# Calling all functions
+processed_data = []
+
+netflix_file = read_netflix_txt(base_dir + 'netflix_movies.txt')
+
+netflix_csv = base_dir + 'netflix_movies.csv'
```

Comment:
Usually output is done to an output directory, not the same directory that the input came from.

I picture that the best way to manage the netflix data is to have it at a known location in the repository (`src/data` is fine) where it is git ignored, and then we should also have a path where the output goes.

Reply:
Just a quick update, as said in another comment `src/data` is probably not good because the data is not technically source code. Probably should be `MediaBridge/data`.

Reply:
Prob ably need a new variable for the output directory, `out_dir`.

And probably should use `os.path.join` instead of string concatenation.

Reply:
See other comment.